### ISSUE TITLE
[BUG] cant use macros in conditions

### ISSUE THREAD

**Issue by amqndin** [WingedSeal/jmc#99]
## Describe the bug
here is the code:
```ts
$if (resist.effect.$(effect_type):@s[type=!player] == true) {
```
and here is the error:
```
JMCValueError
In src/code/arthealth/combat.jmc:91:10
Unrecognized condition 'resist.effect.$' at line 91 col 10.
90 |
91 |    $if (resist.effect.$(effect_type):@s[type=!player] == true) {
             ^^^^^^^^^^^^^^^
92 |      $effect.resist = true;
Consider using one of the following: biome, block, blocks, data, dimension, entity, function, loaded, predicate, score.
```

## To Reproduce

try using a macro thats not wrapped in quotes in a condition

## Desktop

- windows os
- latest jmc build from github as of this day

## Additional context
after some testing its actually the extra bracket in the macro reference that causes this issue.

**Comment by WingedSeal** [comment]
This is a duplicate of https://github.com/WingedSeal/jmc/issues/98

**Comment by Nico314159** [comment]
> This is a duplicate of #98

Personally I would've closed #98 and kept this one open, since #98 is a more specific version of this one.

**Comment by WingedSeal** [comment]
Fixed in https://github.com/WingedSeal/jmc/commit/6829cbfbfaf10b465a0babd40138e168c38c8bc0
